# configuration for youbot base joint interface
youbot_base:
  x: 2
  y: 4
  theta: 0
  position_frame: odom
  base_frame: base_footprint
  velocity_topic: cmd_vel
  
  publish_base_state: true #whether the state of the base is to be published on /joint_states as if it was a joint
  publish_rate: 30 # [Hz]
  base_name: youbot_base #name of the base, joint values are published as three joint values base_name/x base_name/y base_name/theta, following moveit conventions
  
  pid_pos:
    p: 1
    i: 0
    d: 0
  pid_theta:
    p: 1
    i: 0
    d: 0
  
  use_velocity_limits: true  
  unfeasible_rounding_threshold: 
    linear: 0.006 # equal or above gets rounded up to the lowest feasible velocity, everything lower becomes zero
    angular: 0.03
  velocity_limits:
    upper:
      x:
        min: -0.8
        max: 0.8
      y:
        min: -0.8
        max: 0.8
      theta:
        min: -2.86
        max: 2.86
    unfeasible:
      x:
        neg: -0.01
        pos: 0.01
      y:
        neg: -0.01
        pos: 0.01
      theta:
        neg: -0.067
        pos: 0.067
      
    
  use_space_limits: true #limit the space in which the youbot may move
  space_limits:
    x:
      min: -5
      max: 5
    y:
      min: -5
      max: 5